Revit_Core_Engine/Create/View/ViewElevations.cs

```diff
@@ -0,0 +1,108 @@
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using System;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using System.Linq;
+using Autodesk.Revit.UI;
+using BH.oM.Adapters.Revit.Elements;
+using BH.oM.Graphics;
+using ViewPlan = Autodesk.Revit.DB.ViewPlan;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Creates and returns a new Plan view in the current Revit file.")]
+        [Input("document", "Revit current document to be processed.")]
+        [Input("level", "The level that the created Floor Plan refers to.")]
+        [Input("viewName", "Name of the new view.")]
+        [Input("viewFamily", "View family type of the new view.")]
+        [Input("cropBox", "Optional, the crop region to attempt to apply to the newly created view.")]
+        [Input("viewTemplateId", "Optional, the View Template Id to be applied in the view.")]
+        [Input("viewDetailLevel", "Optional, the Detail Level of the view.")]
+        [Input("cropRegionVisible", "True if the crop region should be visible.")]
+        [Input("annotationCrop", "True if the annotation crop should be visible.")]
+        [Output("newView", "The new view.")]
+        public static List<ViewSection> ViewElevations(this Document document, string elevationName, XYZ elevationMarkerLocation, View referenceViewPlan, int numberOfElevations = 4, ElementId viewTemplateId = null, bool cropRegionVisible = false, bool annotationCrop = false)
+        {
+            if (numberOfElevations < 1 || numberOfElevations > 4)
+            {
+                BH.Engine.Base.Compute.RecordWarning($"Number of elevation views should be between 1 and 4.");
```

Comment:
Ahh ok, got it - I would then consider adding an overload to generate a marker plus single view, i.e. `ViewElevation`, to unlock the cases where the walls are not orthogonal. Or would that not make sense?

Reply:
It is not possible to create elevation view without the elevation marker. For not orthogonal walls created elevation views should be rotated but there still must be maximum 4 of them.

Reply:
Yeah, so as discussed offline, I would consider adding a method that would create a marker + single view, i.e. for a 5-walled room you would have 5 markers and 5 elevations, one per each. Not sure if that makes sense from practical point of view, so please dismiss if it does not make sense 😃

Reply:
I think it's necessary to create 1 marker per wall because many rooms have more than 4 walls.

Reply:
Ok, I think have a solution for this - I will rename this method to `Create.ElevationMarker` and it will always create one marker with up to 4 elevation views. For more than 4 walls or other non-rectangular rooms we should use this method multiple times - it depends on the case we have. What do you think @pawelbaran @vietle-bh?

Reply:
How about we first check if the room boundaries form a rectangle? If yes, we create 1 marker for all 4 elevations. If not, we create 1 marker per wall, each has only 1 elevation on it :bowtie:

Reply:
> How about we first check if the room boundaries form a rectangle? If yes, we create 1 marker for all 4 elevations. If not, we create 1 marker per wall, each has only 1 elevation on it :bowtie:

I think that check will add unnecessary overhead particularly when calling the method multiple times.

Reply:
I think there should be no `Room` or `Space` outline input in this method.